Firebase error handling while creating user Swift
I am currently developing an iOS application and I wanted to implement the Firebase database. I'm facing an issue while handling errors during the users creation process. I always get the default error from the switch, here is my code : FIRAuth.auth()?.createUser(withEmail: emailField.text!, password: passwordField.text!, completion: { (user, error) in if (error != nil) { if let errCode = FIRAuthErrorCode(rawValue: error!._code) { var alertController = UIAlertController(title: "", message: "", preferredStyle: UIAlertControllerStyle.alert) let okButton = UIAlertAction(title: "OK", style: UIAlertActionStyle.default) { (result: UIAlertAction) -> Void in print("Error transmitted") } switch errCode { case .errorCodeInvalidEmail: print("Invalid email") alertController = UIAlertController(title: "Error", message: "Email syntax is not correct", preferredStyle: UIAlertControllerStyle.alert) alertController.addAction(okButton) self.present(alertController, animated: true, completion: nil) case .errorCodeEmailAlreadyInUse: print("Email already in use") alertController = UIAlertController(title: "Error", message: "This email is already in use", preferredStyle: UIAlertControllerStyle.alert) alertController.addAction(okButton) self.present(alertController, animated: true, completion: nil) case .errorCodeWeakPassword: print("Password weak") alertController = UIAlertController(title: "Error", message: "Password is too weak. Please choose a password which contains at least 6 characters.", preferredStyle: UIAlertControllerStyle.alert) alertController.addAction(okButton) self.present(alertController, animated: true, completion: nil) default: // ALWAYS GET HERE. print(error) alertController = UIAlertController(title: "Error", message: "An unknown error occured.", preferredStyle: UIAlertControllerStyle.alert) alertController.addAction(okButton) self.present(alertController, animated: true, completion: nil) } } } else { print("User created") let newUser = ["email": self.emailField.text!] let firebaseNewUser = self.ref.childByAutoId() firebaseNewUser.setValue(newUser) } Moreover, the print(error) displays : Optional(Error Domain=FIRAuthErrorDomain Code=17999 "An internal error has occurred, print and inspect the error details for more information." UserInfo={NSUnderlyingError=0x170257be0 { Error Domain=FIRAuthInternalErrorDomain Code=3 "(null)" UserInfo={FIRAuthErrorUserInfoDeserializedResponseKey={ code = 400; errors = ( { domain = usageLimits; message = "Bad Request"; reason = keyExpired; } ); message = "Bad Request"; }}}, error_name=ERROR_INTERNAL_ERROR, NSLocalizedDescription=An internal error has occurred, print and inspect the error details for more information.}) Can someone help me please ?
Try enabling identity toolkit API in google console.
Actually, (in answer to the previous message), the identity toolkit API was already activated. The problem was coming from the fact that an API key was generated but was restrained to websites requests. Solution : So I just generated another API key, restrained it to iOS applications, downloaded the .plist once again, put it into the project and add the API_KEY in it (API_KEY as an information property and the value as the new generated api key value)
For Created New User Please follow this simple code . Please put this code in Create Button @IBAction function. FIRAuth.auth()?.createUserWithEmail(Username.text!, password: Password.text! , completion: { user , error in if error != nil { print("Something Went to Wrong :(") } else { print("Account Created Sucessfully ") } })